scripts/pr-convergence.py

```
@@ -410,6 +439,109 @@ def graphql_open_pulls_by_head_ref(self, head_ref_name: str) -> list[dict[str, A
+        sibling_prs = client.open_pulls_by_base_ref(repo, base_ref)
+    except GhReadError as exc:
+        errors.append({"source": "sibling_prs", "message": str(exc)})
+        return result
+
+    # Keep only open PRs that genuinely target the same base and are neither
+    # this PR nor its own head. The base ref is re-verified from the returned
+    # payload rather than trusted from the request filter, so a server-side
+    # filter change cannot silently widen the candidate set.
+    siblings = []
+    for candidate in sibling_prs:
+        candidate = as_dict(candidate)
```

Ruff (0.16.0)

[warning] `for` loop variable `candidate` overwritten by assignment target

(PLW2901)